h2o-core/src/main/java/water/persist/PersistManager.java

```diff
@@ -247,6 +247,11 @@ private static boolean httpUrlExists(String URLName){
    * @return List of matches
    */
   public List<String> calcTypeaheadMatches(String filter, int limit) {
+    filter = filter.trim();
+    if (filter.isEmpty()) {
+      return new ArrayList<>();
```

Comment:
Default behavior is to return nothing if the path is non-existing. As whitespaces or empty path means non-existing, return nothing as well.

Reply:
Collections.emptyList() is better